lib/types/array.js

```diff
@@ -559,7 +559,8 @@ class ArrayPrompt extends Prompt {
       value = utils.reorder(value);
     }
 
-    this.value = multi ? value.map(ch => ch.name) : value.name;
```

Comment:
⚡ only `.name` was ever considered, even if i gave: `choices: [{ name: 'big bad foo', value: 'foo', { name: 'baaazzzzz', value: 'baz' }]`, for example